pyfixest/utils/dev_utils.py

```diff
@@ -1,24 +1,22 @@
 import re
 from typing import Optional, Union
 
+import narwhals as nw
 import numpy as np
 import pandas as pd
-import polars as pl
+from narwhals.typing import IntoDataFrame
 
-DataFrameType = Union[pd.DataFrame, pl.DataFrame]
+DataFrameType = IntoDataFrame
 
 
-def _polars_to_pandas(data: DataFrameType) -> pd.DataFrame:  # type: ignore
-    if not isinstance(data, pd.DataFrame):
-        try:
-            import polars as pl  # noqa: F401
+def _narwhals_to_pandas(data: IntoDataFrame) -> pd.DataFrame:  # type: ignore
+    try:
+        data = nw.from_native(data).to_pandas()
 
-            data = data.to_pandas()
-        except ImportError:
-            raise ImportError(
-                """Polars is not installed. Please install Polars to use it as
-                an alternative."""
-            )
+    except ImportError:
+        raise ImportError(
+            """Unable to use Narwhals to convert the DataFrame to Pandas."""
+        )
```

Comment:
I don't think it's possible for this to return an `ImportError`, as both narwhals and pandas would already be required dependency according to this PR, I think you can just `return nw.from_native(data).to_pandas()`

I'd also suggest
- using `import narwhals.stable.v1 as nw` instead of `import narwhals as nw`
- writing this as
  ```
  nw.from_native(data, eager_or_interchange_only=True).to_pandas()
  ```

This is because defining a lazy-only layer of support (which DuckDB, PySpark, and others) is high on our priorities for 2025, and we may need to change the API slightly. But if you use `stable.v1`, then we promise that what you've written here will keep working

Reply:
Thanks a lot for taking the time to comment @MarcoGorelli !

I made the suggested changes.